static/js/modals/TerminalPayloadForm.tsx

```diff
@@ -0,0 +1,181 @@
+  return (
+    <Modal
+      title="Reconnect terminal"
+      buttonRow={
+        <>
+          <Button className="u-no-margin--bottom" type="button" onClick={close}>
+            Cancel
+          </Button>
+          <Button appearance="positive" onClick={formik.submitForm}>
+            Reconnect
+          </Button>
+        </>
+      }
+    >
+      <Form onSubmit={formik.handleSubmit}>
+        <Input
+          id="command"
+          name="command"
+          label="Command"
+          type="text"
+          required
+          onBlur={formik.handleBlur}
+          onChange={formik.handleChange}
+          error={formik.touched.command ? formik.errors.command : null}
+          value={formik.values.command}
+          stacked
+        />
+        <Input
+          id="user"
+          name="user"
+          label="User id"
+          type="number"
+          onBlur={formik.handleBlur}
+          onChange={formik.handleChange}
+          value={formik.values.user}
+          stacked
+        />
+        <Input
+          id="group"
+          name="group"
+          label="Group id"
+          type="number"
+          onBlur={formik.handleBlur}
+          onChange={formik.handleChange}
+          value={formik.values.group}
+          stacked
+        />
+        {/* hidden submit to enable enter key in inputs */}
+        <input type="submit" hidden />
+        <Row>
+          <Col size={4}>
+            <Label>Environment variables</Label>
+          </Col>
+          <Col size={4}>Key</Col>
+          <Col size={4}>Value</Col>
+        </Row>
+        {formik.values.environment.map((variable, index) => (
+          <Row key={index}>
+            <Col size={4}> </Col>
+            <Col size={4}>
+              <Input
+                name={`environment.${index}.key`}
+                type="text"
+                onBlur={formik.handleBlur}
+                onChange={formik.handleChange}
+                value={formik.values.environment[index].key}
+                stacked
+              />
+            </Col>
+            <Col size={4}>
+              <Input
+                name={`environment.${index}.value`}
+                type="text"
+                onBlur={formik.handleBlur}
+                onChange={formik.handleChange}
+                value={formik.values.environment[index].value}
+                stacked
+              />
+            </Col>
+          </Row>
+        ))}
+        <Row>
+          <Col size={4}></Col>
+          <Col size={8}>
+            <Button onClick={addEnvironmentRow} type="button" hasIcon>
+              <Icon name="plus" />
+              <span>Add variable</span>
+            </Button>
+            <Button
+              onClick={() =>
+                removeEnvironmentRow(formik.values.environment.length - 1)
+              }
+              type="button"
+              hasIcon
+            >
+              <Icon name="minus" />
+              <span>Remove variable</span>
+            </Button>
```

Comment:
This only allows the user to remove the last added variable, which is not ideal UX. What if I add a bunch of N variables and at some point, I realise that I don't want the variable at position N-10 anymore? I would need to copy the last added variable to replace the variable at N-10, and then remove the last entry: not very user-friendly.
I suggest to put an X/bin icon on the RH side of each entry to allow the user to delete that specific one.